### PR TITLE
add golang generator

### PR DESCRIPTION
**Pull request by polderudo**
Hello Igor,
as i am also not very good at python, i created a new generator (golang).
I removed the dependency to github.com/shopspring/decimal and implemented the needed functionality into a small struct.

I'v been running the code through 96000 datasets through the BFM API and at least for 2021 everthings seems to work ok.
Would be cool if you can accept the PR.

### CODE REVIEW

**Comment by jenner** [comment]
Hi Udo, thanks a lot for the code! Would you consider moving the BigDecimal code to an extra file and use it in the genrated code (much like the PHP generator does)?

**Comment by jenner** [comment]
@knipknap do you think there's a way to merge the code from @polderudo with your Go generator? I'd really like to keep just one generator per language.

**Comment by polderudo** [comment]
> Hi Udo, thanks a lot for the code! Would you consider moving the BigDecimal code to an extra file and use it in the genrated code (much like the PHP generator does)?

Hi, sure. Will do that this evening

**Comment by jenner** [comment]
Hi Udo, what is this taxfiles folder for? It doesn't look like it belongs here

**Comment by polderudo** [comment]
Oh my god. Sorry I did not want to push that... I will fix that in the next few days, vacation...
Also, the 2022 golang generator has some problems with the latest PAP. I had to use the python version for correct values. Will also look for this when I'm back.
Sorry that I pushed that to the PR.

**Comment by polderudo** [comment]
Hello Igor, i pushed a version with the files removed. But the generator still has a problem. Starting from 2800 € we get a rounding error from one cent. The python version is working correct.
I will have to investigate where the error comes from.

**Comment by polderudo** [comment]
Hello Igor, i fixed the problem with the rounding in the Divide function. Also the 2022 PAP is in and i also tested it through the BFM API. If you like you could merge the PR as new version with the golang generator. Seems, like we get no response from the go guy.

File: lstgen/cli.py [hunk 1]
```diff
@@ -10,9 +10,9 @@
 
 from lxml import etree
 
-from . import PapParser
-from . import pap
-from .generators import GENERATORS
+from lstgen import PapParser
+from lstgen import pap
+from lstgen.generators import GENERATORS
```
Comment (jenner) [inline lstgen/cli.py:15]: Any particular reason for this change?

File: lstgen/cli.py [hunk 2]
```diff
@@ -10,9 +10,9 @@
```
Comment (jenner) [inline lstgen/cli.py:10]: I don't think this belongs here

File: setup.py [hunk 3]
```diff
@@ -23,7 +23,7 @@
 ]
 
 setup(name='LstGen',
-    version='0.5.4',
+    version='0.5.6',
```
Comment (jenner) [inline setup.py:26]: Please let me change this in an extra commit when I bump the version and update the release note

File: lstgen/generators/golang/__init__.py [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+            instance=self.instance_var,
+            cls=self.class_name,
+            name=method.name
+        )
+        # actual method body
+        with self.writer.indent(signature):
+            self._write_stmt_body(method)
+
+    def convert_to_go(self, value):
+        """ Converts java pseudo code into valid java code """
```
Comment (jenner) [inline lstgen/generators/golang/__init__.py:291]: Misleading docstring :)

**Review by jenner — CHANGES_REQUESTED** [review]

**Comment by polderudo** [comment]
Hello Igor, i pushed a new version. Sorry, my python skills are not that good...

**Comment by jenner** [comment]
Thanks Udo! Do you think we can safely remove the "old" go generator? I think it makes little to no sense to have two of them

**Comment by polderudo** [comment]
> Thanks Udo! Do you think we can safely remove the "old" go generator? I think it makes little to no sense to have two of them

I think yes. I mean, it did not round correct anyway, so even if my version would have bugs it would not be worse :)

**Pull request merged**